src/App.js

```diff
@@ -3,18 +3,51 @@ import React from "react";
 
 const IMAGE_HEIGHT = 200;
 
-class Image extends React.Component {
+class ImageList extends React.Component {
+
+    render() {
+        const cats = this.props.cats;
+        return (
+            <div>
+                {cats}
+            </div>
+        );
+    }
+}
+
+
+class InputForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(e) {
+        this.props.onCountChange(e.target.value);
+    }
+
+    render() {
+        const count = this.props.count;
+        return (
+            <fieldset>
+                <legend>Enter count</legend>
+                <input type="number" value={count}
+                       onChange={this.handleChange}/>
+            </fieldset>
+        );
+    }
+}
+
+
+class App extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
+            count: 0,
```

Comment:
You are duplicating the state again. There is no need to store the count of the images separately. You should use just `this.state.images.length` instead.